### ISSUE TITLE
Classpath resources are not correctly copied

### ISSUE THREAD

**Issue by SomeoneToIgnore** [vaadin/flow#3315]
@denis-anisimov commented on [Fri Jan 12 2018](https://github.com/vaadin/flow-maven-plugin/issues/46)

I'm using bakery app (latest commit is 93f120fab1c139100593f087c5ff0226a3521d3d ).

I've modified the `pom.xml` file with the patch (see below, for some reasons github is unable to upload it).
So what I did: 
- removed two sections with copy and yarn 
- replaced them with flow maven plugin

May be it still requires some additional configuration (I'm not sure).

But here what I git in the result of transpilation:

```
 class FlowComponentRenderer extends Polymer.Element {
[INFO]                                       ~~~~~~~~~~~~~~~
[INFO] 
[INFO] frontend/flow-component-renderer.html(5,38) warning [unknown-superclass] - Unable to resolve superclass Polymer.Element
[INFO] 
[INFO] 
[INFO]     class GridFlowSelectionColumnElement extends Vaadin.GridColumnElement {
[INFO]                                                  ~~~~~~~~~~~~~~~~~~~~~~~~
[INFO] 
[INFO] frontend/vaadin-grid-flow-selection-column.html(22,49) warning [unknown-superclass] - Unable to resolve superclass Vaadin.GridColumnElement
```

`flow-component-renderer.html` if a file from the `flow-server` resources folder `src/main/resources/META-INF/resources/frontend/flow-component-renderer.html`

If you look into it has an import statement at the very beginning : `
<link rel="import" href="bower_components/polymer/polymer.html">``

The file `frontend/flow-component-renderer.html`  ( which is a copy of it I suppose) doesn't have this import and the very first line inside it is: 

`<dom-module id="vaadin-grid-flow-selection-column">`

And I suppose that's the reason of the messages in the report: it can't resolve imported class.

Unfortunately I don't know all the details of plugin functionality and can't say whether this is important or not.
At least the build freezes for a long time and then prints those messages.
But in the end they don't prevent the further build execution.

The command line which I use is : `mvn -Dvaadin.productionMode=true install`

```
diff --git a/pom.xml b/pom.xml
index f782b74..3aa4fc0 100644
--- a/pom.xml
+++ b/pom.xml
@@ -84,6 +84,13 @@
 
     </repositories>
 
+    <pluginRepositories>
+        <pluginRepository>
+            <id>vaadin-prereleases</id>
+            <url>https://maven.vaadin.com/vaadin-prereleases</url>
+        </pluginRepository>
+    </pluginRepositories>
+
     <dependencyManagement>
         <dependencies>
             <dependency>
@@ -371,97 +378,23 @@
                     <artifactId>flow-server-production-mode</artifactId>
                 </dependency>
             </dependencies>
-
+            
             <build>
                 <plugins>
-                    <!-- (3): Unpack those frontend project dependencies that are embedded into Flow and Flow components
-                         JARs so that they can be included into the production frontend build. This step will become
-                         unnecessary after switching to the flow-maven-plugin -->
                     <plugin>
-                        <groupId>org.apache.maven.plugins</groupId>
-                        <artifactId>maven-dependency-plugin</artifactId>
-                        <version>3.0.0</version>
+                        <groupId>com.vaadin</groupId>
+                        <artifactId>flow-maven-plugin</artifactId>
+                        <version>0.0.5</version>
                         <executions>
                             <execution>
-                                <id>unpack-flow-component-renderer-html</id>
-                                <phase>process-sources</phase>
                                 <goals>
-                                    <goal>unpack</goal>
+                                    <!-- specify goals required here -->
+                                    <goal>copy-production-files</goal>
+                                    <goal>transpile-production-files</goal>
                                 </goals>
+                                <!-- specify goals' configuration here, can be omitted, if 
+                                    defaults are used -->
                                 <configuration>
-                                    <artifactItems>
-                                        <!-- flow-component-renderer.html is a dependency of `ComponentTemplateRenderer` -->
-                                        <artifactItem>
-                                            <groupId>com.vaadin</groupId>
-                                            <artifactId>flow-server</artifactId>
-                                            <type>jar</type>
-                                            <overWrite>true</overWrite>
-                                            <includes>META-INF/resources/frontend/flow-component-renderer.html</includes>
-                                            <outputDirectory>${project.build.directory}</outputDirectory>
-                                        </artifactItem>
-
-                                        <!-- gridConnector.js is a dependency of `Grid<T>` -->
-                                        <artifactItem>
-                                            <groupId>com.vaadin</groupId>
-                                            <artifactId>vaadin-grid-flow</artifactId>
-                                            <type>jar</type>
-                                            <overWrite>true</overWrite>
-                                            <includes>META-INF/resources/frontend/gridConnector.js</includes>
-                                            <outputDirectory>${project.build.directory}</outputDirectory>
-                                        </artifactItem>
-                                    </artifactItems>
-                                </configuration>
-                            </execution>
-                        </executions>
-                    </plugin>
-                    <!-- (4): copy the JAR-embedded frontend dependencies into the location where they can be picked up
-                         by the production frontend build -->
-                    <plugin>
-                        <groupId>org.apache.maven.plugins</groupId>
-                        <artifactId>maven-resources-plugin</artifactId>
-                        <executions>
-                            <execution>
-                                <id>copy-flow-component-renderer-html</id>
-                                <phase>process-sources</phase>
-                                <goals>
-                                    <goal>copy-resources</goal>
-                                </goals>
-                                <configuration>
-                                    <outputDirectory>${frontend.directory}</outputDirectory>
-                                    <resources>
-                                        <resource>
-                                            <directory>${project.build.directory}/META-INF/resources/frontend</directory>
-                                            <includes>flow-component-renderer.html,gridConnector.js</includes>
-                                        </resource>
-                                    </resources>
-                                </configuration>
-                            </execution>
-                        </executions>
-                    </plugin>
-
-                    <!-- (5): Configures the frontend plugin to compile the web components source -->
-                    <plugin>
-                        <groupId>com.github.eirslett</groupId>
-                        <artifactId>frontend-maven-plugin</artifactId>
-                        <executions>
-                            <execution>
-                                <id>install-node-and-yarn</id>
-                                <goals>
-                                    <goal>install-node-and-yarn</goal>
-                                    <goal>yarn</goal> <!-- runs 'install' by default -->
-                                </goals>
-                                <configuration>
-                                    <workingDirectory>${buildtools.directory}</workingDirectory>
-                                </configuration>
-                            </execution>
-                            <execution>
-                                <id>run-polymer-build</id>
-                                <goals>
-                                    <goal>yarn</goal>
-                                </goals>
-                                <configuration>
-                                    <workingDirectory>${buildtools.directory}</workingDirectory>
-                                    <arguments>${yarn.build.goal}</arguments>
                                 </configuration>
                             </execution>
                         </executions>

```

---

@denis-anisimov commented on [Fri Jan 12 2018](https://github.com/vaadin/flow-maven-plugin/issues/46#issuecomment-357188164)

Also see #45 for the issue with 0.0.5 plugin version (I've modified the plugin to be able to make it work).

**Comment by pleku** [comment]
Note that bakery is importing these templates to hack their current bundling to work, this should not be done when `flow-maven-plugin` is in place:
https://vaadin.slack.com/archives/C6X43FE8M/p1516215851000177?thread_ts=1516204277.000021&cid=C6X43FE8M

**Comment by denis-anisimov** [comment]
Really weird but I'm not able to reproduce this issue anymore.
Instead I see the behavior described in #3367 which is totally different thing.
Now the copied files have exactly the same content as the originals: I see there import statement.

Tried with the latest available plugin version (at the moment): 1.0.0.alpha18.
May be the previous behavior somehow related to the difference between versions hack.
